backend/controllers/passController.js

```diff
@@ -0,0 +1,90 @@
+const asyncHandler = require("express-async-handler")
+require("dotenv").config()
+const { connectDB } = require("../config/db")
+const User = require("../model/user.model")
+const Pass = require("../model/pass.model")
+const { ErrorHandler } = require("../middleware/errorMiddleware")
+
+//@desc Generate a pass by user
+//@route POST /users/generatePass
+const generatePass = asyncHandler(async (req, res, next) => {
+  try {
+    await connectDB()
+    const { name, email, phone, duration, start } = req.body
+
+    const user = await User.findOne({ _id: req.user._id })
+    if (user) {
+      const newPass = await Pass.create({
+        name,
+        phone,
+        email,
+        duration,
+        start,
+        checkInStatus: false,
+        generatedUserId: user.id,
+        userName: req.user.name,
+      })
+      if (newPass) {
+        res.status(201).json({
+          success: true,
+          newPass: newPass,
+          message: "Pass generated successfully",
+        })
+      }
+    } else {
+      throw new ErrorHandler(401, "Signup to create Pass!")
+    }
+  } catch (err) {
+    next(err)
+  }
+})
+
+//@desc View all passes generated by a user
+//@route Get /users/viewPasses
+const viewPasses = asyncHandler(async (req, res, next) => {
+  try {
+    await connectDB()
+    const user = await User.findById(req.user._id)
```

Comment:
Remove this as we are not using it. we can use req.user._id at line 49

Reply:
done

Reply:
sorry, did not saw it yesterday